ads/google/a4a/utils.js

```diff
@@ -121,10 +125,22 @@ export function googleAdUrl(
     const viewportRect = viewport.getRect();
     const iframeDepth = iframeNestingDepth(win);
     const viewportSize = viewport.getSize();
-    if (ValidAdContainerTypes.indexOf(adElement.parentElement.tagName) >= 0) {
-      queryParams.push({name: 'amp_ct',
-                        value: adElement.parentElement.tagName});
-    }
+    // Detect container types.
+    const containerTypeSet = {};
+    closest(adElement.parentElement, el => {
```

Comment:
sorry i didn't notice that you want a full scan of the ancestors. `closest` is not built for this. would you switch back?